metaflow/plugins/argo/argo_events_decorator.py

```diff
@@ -0,0 +1,173 @@
+from metaflow.decorators import FlowDecorator
+from metaflow.exception import MetaflowException
+from metaflow.util import is_stringish
+
+# TODO: Support dynamic parameter mapping through a context object that exposes
+#       flow name and user name similar to parameter context
+
+
+# TODO: This decorator interface can be a top-level decorator to support similar
+#       implementations for Netflix Maestro, AWS Step Functions, Apache Airflow...
+class ArgoEventsDecorator(FlowDecorator):
+    name = "trigger"
+    defaults = {
+        "event": None,
+        "events": [],
+        "options": {},  # TODO: introduce support for options
+    }
+
+    def flow_init(
+        self, flow, graph, environment, flow_datastore, metadata, logger, echo, options
+    ):
+        self.triggers = []
+        if sum(map(bool, (self.attributes["event"], self.attributes["events"]))) > 1:
+            raise MetaflowException(
+                "Specify only one of *event* or *events* "
+                "attributes in *@trigger* decorator."
+            )
+        elif self.attributes["event"]:
+            # event attribute supports the following formats -
+            #     1. event='table.prod_db.members'
+            #     2. event={'name': 'table.prod_db.members',
+            #               'parameters': {'alpha': 'member_weight'}}
```

Comment:
what is the meaning of the dict syntax? Is this for the mapping of parameters?

Reply:
correct. as specified in https://github.com/Netflix/metaflow-docs/pull/86/files